nntrainer/compiler/tflite_interpreter.cpp

```diff
@@ -56,12 +61,135 @@ void builder2file(const flatbuffers::FlatBufferBuilder &builder,
  *
  */
 class TfOpNode {
+public:
+  using Variables = std::vector<const Var_Grad *>;
+
+  TfOpNode() = default;
+
+  /**
+   * @brief Construct a new Tf Op Node object from layer
+   * @note this is a shortcut to skip if layer does not need to be devided or
+   * fused
+   * @param layer layer that is converted to TfOpNode
+   */
+  TfOpNode(const Layer &layer) {
+    setInputs(layer.getInputRef());
+    setOutputs(layer.getOutputRef());
+    setWeights(layer.getWeightsRef());
+    setOpType(layer.getType());
+  }
+
+  /**
+   * @brief Set the Inputs object from layer
+   *
+   * @param inputs_ input to be inserted
+   */
+  void setInputs(const std::vector<std::shared_ptr<Var_Grad>> &inputs_) {
+    inputs.reserve(inputs_.size());
+    std::transform(inputs_.begin(), inputs_.end(), std::back_inserter(inputs),
+                   [](const auto &data) { return data.get(); });
```

Comment:
This code will create a new Var_Grad object for every element in the vector. Is this intended?

Reply:
This is weak reference to the function. else I can't say they are equivalent.  The data structure does not own the pointer. Also, I think this case is fairly safe as it should be guaranteed that the given graph doesn't get destroyed while saving.

If you urge not to use, can you give me some alternatives for this purpose?

Reply:
I meant a templated common function which these functions can call. Feel free to do in a later PR.

Reply:
> but if someone re-uses this later, this will be a hard bug to find.

I think putting this to shared_ptr hide bugs that model got destroyed but the opNode still works.

Another problem is that weights are not shared_ptr<Var_Grad>, I can't get the original reference of the weights other than this. If this is resolved, I can change to shared_ptr but for now shared_ptr can't be used.

Yet another problem is each `Var_Grad *` is used as a hash key to the index map. So index map will anyway going to have the raw pointer managing inside. (Defining a hash function for a shared_ptr is possible, but IMO this is an overkill for this stage, but later polished.)

Reply:
We discussed offline that this is (kind of) resolved using explicit ownership through `unique_ptr` in the upcoming PR.